ToolChain/Assembler/Parser/Parser.py

```diff
@@ -20,586 +19,393 @@
-        self._evaluateAndExtractInfoFromLine(line, buildInstruction, rawLine)
-
-        if type(buildInstruction.instructionCode) is not str:
-            # We assume it's an instruction (integer), since memory references or comments would be strings
-            # Parser will verify all possible codes, and generate the appropriate instruction based on its form
-            possibleCodes = self._findPossibleCodesForInstruction(buildInstruction)
-            instructionCode, instructionForm = self._getInstructionCodeAndFormUsingPossibleCodes(buildInstruction,
-                                                                                                 possibleCodes)
-            if instructionForm is None or instructionCode is None:
-                raise ValueError("Assembler general error")
-            else:
-                # Code is valid, fetch the instruction's form based on the arguments, then generate the instruction
-                buildInstruction.instructionCode = instructionCode
-                buildInstruction.instructionLength = formDescription[instructionForm]["length"]
-                lineInstruction = self._generateBinaryInstructionFromInstruction(buildInstruction,
-                                                                                 formDescription[instructionForm])
         else:
-            # In this case, there were no instructions. Thus, we look at other possibilities
-            if COMMENT_INDICATORS in buildInstruction.instructionCode:
-                # Simply ignore a comment line
-                buildInstruction = None
-                lineInstruction = None
-
-            #  In these next cases, we can treat them as "data" variables with no real instructions.
-            #  We store the data as the instruction's source immediate value, and set instruction code to 0
-            elif DATA_ALPHA_INDICATOR in buildInstruction.instructionCode:
-                # This is a bunch of alpha data (String)
-                lineInstruction = buildInstruction.sourceImmediate
-                buildInstruction.instructionCode = 0
-
-            elif DATA_NUMERIC_INDICATOR in buildInstruction.instructionCode:
-                # This is a numeric data field (Integer)
-                lineInstruction = struct.pack(">I", buildInstruction.sourceImmediate)
-                buildInstruction.instructionCode = 0
-
-            elif DATA_MEMORY_REFERENCE in buildInstruction.instructionCode:
-                # This is a data memory reference (dataMemref)
-                lineInstruction = buildInstruction.sourceImmediate
-                buildInstruction.instructionCode = 0
-
-            else:
-                # Having exhausted all options, we assume there was nothing else on this line.
-                lineInstruction = None
+            # The instruction was not in the list, and no identifier mnemonics were found.
+            raise ValueError("Invalid code format at line " + str(self.lineno))
```

Comment:
I would change error message to "Invalid assembly instruction format..." or something similar to make it a bit more precise.